backend/apps/recommendations/services/github_analyzer.py

```
@@ -0,0 +1,258 @@
+            'total_repos': len(repos),
+            'languages': languages,
+            'skill_levels': skill_levels,
+            'total_commits': commits,
+            'frameworks': self._extract_frameworks(repos),
+            'contributions': self._get_contributions(username),
+        }
+    
+    def _get_user_data(self, username: str) -> Optional[Dict]:
+        """Get user data from GitHub API."""
+        try:
+            response = requests.get(
```

Ruff (0.15.20)

[error] Probable use of `requests` call without timeout

(S113)